godot-macros/src/class/derive_godot_class.rs

```diff
@@ -349,22 +350,31 @@ fn parse_struct_attributes(class: &venial::Struct) -> ParseResult<ClassAttribute
     })
 }
 
-/// Returns field names and 1 base field, if available
-fn parse_fields(class: &venial::Struct, init_strategy: InitStrategy) -> ParseResult<Fields> {
-    let mut all_fields = vec![];
-    let mut base_field = Option::<Field>::None;
-    let mut has_deprecated_base = false;
-
-    let named_fields: Vec<(venial::NamedField, Punct)> = match &class.fields {
-        venial::Fields::Unit => {
-            vec![]
-        }
+/// Fetches data for all named fields for a struct
+///
+/// Errors if `class` is a tuple struct
+fn named_fields(class: &venial::Struct) -> ParseResult<Vec<(venial::NamedField, Punct)>> {
+    // This needs to be separate from parse_fields because it needs to
+    // run before parse_struct_attributes since error from here demands
+    // a larger refactor than from there, so should be seen first.
```

Comment:
Maybe split this sentence, it's a bit hard to understand. And feel free to use longer lines :wink:

Reply:
What width do you prefer?  This is already wider than the 72 characters I normally use

Reply:
Oh excuse me, I always forget that my editor opens narrower than 80 chars.  I need to adjust that.

Reply:
i believe we use 120-145.

Reply:
Exactly, https://godot-rust.github.io/book/contribute/conventions.html: 

> Line width is 120-145 characters (mostly relevant for comments).

Reply:
Gotcha :+1: